node/src/client.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RuntimeApi for client
+
+use manta_primitives::types::{AccountId, Balance, Block, Index as Nonce};
+use session_key_primitives::{AuraId, NimbusId};
+use sp_runtime::traits::BlakeTwo256;
+
+/// RuntimeApiCommon + RuntimeApiNimbus: nimbus
+///
+/// Common RuntimeApi trait bound
+pub trait RuntimeApiCommon:
+    sp_api::Metadata<Block>
+    + sp_api::ApiExt<Block>
+    + sp_block_builder::BlockBuilder<Block>
+    + sp_offchain::OffchainWorkerApi<Block>
+    + sp_session::SessionKeys<Block>
+    + sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>
+where
+    <Self as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
+{
+}
+
+/// Extend RuntimeApi trait bound for Nimbus
+pub trait RuntimeApiNimbus:
+    pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance>
+    + frame_rpc_system::AccountNonceApi<Block, AccountId, Nonce>
+    + cumulus_primitives_core::CollectCollationInfo<Block>
+    + sp_consensus_aura::AuraApi<Block, AuraId>
+    + nimbus_primitives::AuthorFilterAPI<Block, NimbusId>
+    + nimbus_primitives::NimbusApi<Block>
+{
+}
+
+impl<Api> RuntimeApiCommon for Api
+where
+    Api: sp_api::Metadata<Block>
+        + sp_api::ApiExt<Block>
+        + sp_block_builder::BlockBuilder<Block>
+        + sp_offchain::OffchainWorkerApi<Block>
+        + sp_session::SessionKeys<Block>
+        + sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>,
+    <Self as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
+{
+}
+
+impl<Api> RuntimeApiNimbus for Api where
+    Api: pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance>
+        + frame_rpc_system::AccountNonceApi<Block, AccountId, Nonce>
+        + cumulus_primitives_core::CollectCollationInfo<Block>
+        + sp_consensus_aura::AuraApi<Block, AuraId>
```

Comment:
technically `sp_consensus_aura` is not a NimbusAPI but instead left-over from Aura consensus.
Since you removed the Aura one, i'd leave this one line outside of the wrapper types

Reply:
like you did in line 121
https://github.com/Manta-Network/Manta/pull/836/files#diff-594440db8adfc5d1b37711309312bfa06b49e488309186ad062a896778cf9fbfR121

Reply:
I'm also moving `TransactionPaymentRuntimeApi` and `AccountNonceApi` to commonn part, this will make `new_partial` more strict bound then before, But I think it's ok.
PS: can we also move `AuraApi` to common part too, in that case we dont' need add extra AuraApi in service.

Reply:
> PS: can we also move AuraApi to common part too, in that case we dont' need add extra AuraApi in service.

Manta won't have the aura+nimbus consensus when it launches so it won't need the Aura API, so IMO Aura shouldn't be in common

Reply:
oh right, that code hasn't been updated yet - it will be before launch, but for now it's needed to compile.
switching this over would take longer than today so leave `service_aura` in for now